src/main/java/net/ossindex/gradle/OssIndexPlugin.java

```diff
@@ -45,27 +52,74 @@ public void apply(Project project) {
         audit.doLast(this::doAudit);
     }
 
+    /**
+     * Get the proxy if it exists. Check 3 different places:
+     *
+     *   1. The build.gradle file
+     *   2. The local properties
+     *   3. The system properties
+     */
     private Proxy getProxy(Project project, String scheme) {

```

Comment:
It seems like the system properties gets the highest precedence in defining a proxy, because the `proxy` is `new`ed in every if branch, is this intended?
I would expect the `build.gradle` to have the highest precedence, then the local and then system.

That means that if I specify all 3, I want to get the one from the `build.gradle`.

But maybe it is intended this way :-)

Reply:
Oh I didn't catch the `!proxy.isValid()` in the other cases. Makes sense now :)